tests/Functional/Application/ReadySession/GetReadySessionQueryHandlerTest.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+namespace StepupReadId\Tests\Functional\Application\ReadySession;
+
+use StepupReadId\Application\ReadySession\GetReadySessionQuery;
+use StepupReadId\Application\ReadySession\GetReadySessionQueryHandler;
+use StepupReadId\Domain\ReadySession\Model\ReadySession;
+use StepupReadId\Domain\ReadySession\Model\ReadySessionTTL;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Symfony\Component\HttpClient\MockHttpClient;
+use Symfony\Component\HttpClient\Response\MockResponse;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use function json_encode;
+
+class GetReadySessionQueryHandlerTest extends KernelTestCase
+{
+    /** @var GetReadySessionQueryHandler */
+    private static $getReadySessionQueryHandler;
+
+    public static function setUpBeforeClass(): void
+    {
+        self::bootKernel();
+
+        $container = self::$kernel->getContainer();
+        $container->set('test.' . HttpClientInterface::class, new MockHttpClient(static::responses()));
+
+        static::$getReadySessionQueryHandler = $container->get('test.' . GetReadySessionQueryHandler::class);
+    }
+
+    public function testRequestReadySession(): void
+    {
+        $response = (static::$getReadySessionQueryHandler)(new GetReadySessionQuery(ReadySessionTTL::MINIMUM_TTL));
```

Comment:
Not that intuitive from a readability perspective. But certainly gets the job done!

Reply:
Yes  I know. But I have to use static calls because the way mocking http responses are called and configured. I can change to this if you think is more cleaner:

```
static::$getReadySessionQueryHandler->__invoke(new GetReadySessionQuery(ReadySessionTTL::MINIMUM_TTL));
```

Reply:
That does make the use of the `__invoke` magic method more verbose. I like it.